File: Jumpeno.Client/wwwroot/js/document/JSNavigator.js
class JSNavigator {
    // State ------------------------------------------------------------------------------------------------------------------------------
    static State = key => window.history.state?.[key] || null

    static SetState = (key, state, url) => window.history.replaceState({ ...window.history.state, [key]: state}, "", url)
}

window.JSNavigator = JSNavigator;

File: Jumpeno.Client/Services/JavaScript/Constants/document/JSNavigator.cs
namespace Jumpeno.Client.Constants;

public static class JSNavigator {
    public static readonly string ClassName = typeof(JSNavigator).Name;

    // State ------------------------------------------------------------------------------------------------------------------------------
    public static readonly string State = $"{ClassName}.State";
    public static readonly string SetState = $"{ClassName}.SetState";

    // Media ------------------------------------------------------------------------------------------------------------------------------
    public static readonly string IsTouchDevice = $"{ClassName}.IsTouchDevice";
}
